layer/event_logging.h

```diff
@@ -84,6 +87,43 @@ class AttributeImpl : public Attribute {
   T value_;
 };
 
+// The base class for an attribute that keeps the duration information. To take
+// the time unit into account, the subclasses must implement the `GetValue()`
+// method such that it converts the underlying `duration` value to a specific
+// unit(nanoseconds, miliseconds, seconds, etc).
+class DurationAttr : public Attribute {
+ public:
+  static constexpr ValueType id_ = ValueType::kDuration;
+
+  virtual const int64_t &GetValue() const = 0;
```

Comment:
Why not return the underlying duration?

Reply:
And additionally, we can add more getter to get the number of nanoseconds etc.

Reply:
> Why not return the underlying duration?

I wanted to treat the value returned by this function as a value that doesn't need an extra step of conversion so that the logger doesn't need to be concerned about the time unit.

Reply:
Oh, I see. So you want the dynamic type of the duration to determine the printing format. In this case, you would need a new ID for each duration unit.

Reply:
Edit: I think I get it know, now, this is what's already going on, except that DurationNsAttr doesn't have it's own unit.
This seems fine to me then, just return `int64_t` by value.